components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +132,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.where(column, OperatorType.LIKE, "%");
```

Comment:
Another observation. With table like this:  
![messages_varchar2](https://cloud.githubusercontent.com/assets/13213915/17438015/0f3069ac-5b21-11e6-8fae-008824b32ec0.png)

```
SELECT "MESSAGES_VARCHAR2"."ID", "MESSAGES_VARCHAR2"."MESSAGE" 
FROM DATACLEANER."MESSAGES_VARCHAR2" 
WHERE "MESSAGES_VARCHAR2"."ID" IS NOT NULL AND "MESSAGES_VARCHAR2"."MESSAGE" IS NOT NULL
```

returns expected 1. row. 

However, when AND "MESSAGES_VARCHAR2"."MESSAGE" <> ''' is added:
```
SELECT "MESSAGES_VARCHAR2"."ID", "MESSAGES_VARCHAR2"."MESSAGE" 
FROM DATACLEANER."MESSAGES_VARCHAR2" 
WHERE "MESSAGES_VARCHAR2"."ID" IS NOT NULL AND "MESSAGES_VARCHAR2"."MESSAGE" IS NOT NULL AND "MESSAGES_VARCHAR2"."MESSAGE" <> ''
```
It returns nothing.